integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/search/predicate/QueryStringPredicateSpecificsIT.java

```diff
@@ -281,94 +281,130 @@ void queryStringSyntax(String field, String value1, String value2, String noMatc
-						.fetchAllHits()
-		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "1", "2" );
-
-		assertThatHits(
-				scope.query()
-						.where( f -> f.queryString().field( field )
-								.matching( String.format( Locale.ROOT, "{%s TO %s}", value1, value2 ) ) )
-						.fetchAllHits()
-		).isEmpty();
-
-		assertThatHits(
-				scope.query()
-						.where( f -> f.queryString().field( field )
-								.matching( String.format( Locale.ROOT, "[%s TO %s}", value1, value2 ) ) )
-						.fetchAllHits()
-		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "1" );
-
-		assertThatHits(
-				scope.query()
-						.where( f -> f.queryString().field( field )
-								.matching( String.format( Locale.ROOT, "{%s TO %s]", value1, value2 ) ) )
-						.fetchAllHits()
-		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "2" );
-
-		assertThatHits(
-				scope.query()
-						.where( f -> f.queryString().field( field )
-								.matching( String.format( Locale.ROOT, "[%s TO *]", value1 ) ) )
-						.fetchAllHits()
-		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "1", "2", "3" );
-
-		assertThatHits(
-				scope.query()
-						.where( f -> f.queryString().field( field )
-								.matching( String.format( Locale.ROOT, "{%s TO *]", value1 ) ) )
-						.fetchAllHits()
-		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "2", "3" );
-
-		assertThatHits(
-				scope.query()
-						.where( f -> f.queryString().field( field )
-								.matching( String.format( Locale.ROOT, "[* TO %s]", value2 ) ) )
-						.fetchAllHits()
-		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "1", "2" );
-
-		assertThatHits(
-				scope.query()
-						.where( f -> f.queryString().field( field )
-								.matching( String.format( Locale.ROOT, "[* TO %s}", value2 ) ) )
-						.fetchAllHits()
-		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "1" );
-
-		assertThatHits(
-				scope.query()
-						.where( f -> f.queryString().field( field ).matching( "-" + noMatch ) )
-						.fetchAllHits()
-		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "1", "2", "3" );
-		assertThatHits(
-				scope.query()
-						.where( f -> f.queryString().field( field ).matching( "!" + noMatch ) )
-						.fetchAllHits()
-		).hasDocRefHitsAnyOrder( indexForSyntaxParsingCheck.typeName(), "1", "2", "3" );
-
-		assertThatHits(
-				scope.query()
-						.where( f -> f.queryString().field( field ).matching( noMatch ) )
-						.fetchAllHits()
-		).isEmpty();
+		//   We are not running these range tests if the value contains square brackets [] because there is a problem in a Lucene query parser
+		//   and even if the query is escaped as `[2018\-01\-01T12\:58\:30.000000000\+02\:00\[Africa\/Cairo\] TO 2018\-02\-01T08\:15\:30.000000000\-02\:00\[\-02\:00\]]`
+		//   Fails for both Lucene and Elasticsearch backends...
+		if ( !( value1.contains( "]" ) || value2.contains( "]" ) ) ) {
```

Comment:
note this "condition" 😃 
while the range queries do work with ZDT fields, as soon as there's that optional text part in brackets parser fails to parse the query. it fails for Elasticsearch as well. 
I've tried looking into the parser internals, but 😨 🙈

Reply:
Makes sense. About the escape, do you mean that `\` is indeed intended for escapes but doesn't work, or is it just that `\` isn't supported? The escape character might be different...

Reply:
BTW I hope our parser for `ZonedDateTime` is lenient then... e.g. does it accept something with just an offset, but no timezone?
For that matter, does it accept a date without a time (but with an offset)? Those are the formats that end users are more likely to use...

Reply:
https://lucene.apache.org/core/2_9_4/queryparsersyntax.html#Escaping%20Special%20Characters

don't look at the version 🙈 
but yeah it is suppose to escape special characters, and it does in all other cases, otherwise `-10` is treated as `not 10`. but when it comes to range query string it seems theres' a problem in the lucene's parser, and it doesn't correctly escape the brackets

Reply:
Okay then, probably something to report to Lucene. But not a blocker for us.

Reply:
> BTW I hope our parser for ZonedDateTime is lenient then... e.g. does it accept something with just an offset, but no timezone?

We use  this: 
```java
ISO_ZONED_DATE_TIME = new DateTimeFormatterBuilder()
                .append(ISO_OFFSET_DATE_TIME)
                .optionalStart()
                .appendLiteral('[')
                .parseCaseSensitive()
                .appendZoneRegionId()
                .appendLiteral(']')
                .toFormatter(ResolverStyle.STRICT, IsoChronology.INSTANCE);
```
and I've tried the strings with just offset and it's fine, I even tried to do `toOffsetDateTime().toZoneDateTime()` in these tests to get rid of the zone, but then there were problems with Elasticsearch and I didn't want to deal with those 🙈

Reply:
> but then there were problems with Elasticsearch and I didn't want to deal with those 🙈

Ok. You might want to keep them in mind though, because that possibly means we should add more "input" formats for our ZonedDateTime field types. Or just solve https://hibernate.atlassian.net/browse/HSEARCH-2354.